src/main/java/org/jsoup/parser/HtmlTreeBuilder.java

```diff
@@ -282,6 +282,14 @@ Element insertEmpty(Token.StartTag startTag) {
     }
 
     FormElement insertForm(Token.StartTag startTag, boolean onStack, boolean checkTemplateStack) {
+        // cleanup duplicate attributes:
```

Comment:
I think that we should factor this out so the code is not duplicated on 230. Either a common method, or better, a shared code path.

Reply:
Sorry. I already added comment below